serve.go

```diff
@@ -24,19 +27,46 @@ func (s *Server) Run() error {
 // It also generates the OpenAPI spec and outputs it to a file, the UI, and a handler (if enabled).
 func (s *Server) RunTLS(certFile, keyFile string) error {
 	s.isTLS = true
-
-	s.setup()
-	return s.Server.ListenAndServeTLS(certFile, keyFile)
+	if err := s.setup(); err != nil {
+		return err
+	}
+	return s.Server.ServeTLS(s.listener, certFile, keyFile)
 }
 
-func (s *Server) setup() {
+func (s *Server) setup() error {
+	if err := s.setupDefaultListener(); err != nil {
+		return err
+	}
```

Comment:
I think you should set the `Addr` here.

Like where you do this in the `WithListener` function

```golang
WithAddr(listener.Addr().String())(s)
```

That way we ensure the port being printed in the startup message is for sure correct. At this point there is no way the user can potentially overwrite it. At least our final feedback would be correct.